File: src/common/web/data/entities/connector/categories/RepositoryConnectorCategory.ts
import { ConnectorCategory, type ConnectorCategoryID } from "./ConnectorCategory";

export class RepositoryConnectorCategory extends ConnectorCategory {
    public static readonly CategoryID: ConnectorCategoryID = "repository";

    public constructor() {
        super("Repository", "Publish your project to make it available to the public.");
    }
}

File: src/common/py/data/entities/connector/categories/repository_connector_category.py
import typing
from dataclasses import dataclass

from .connector_category import ConnectorCategory, ConnectorCategoryID
from .connector_categories_catalog import ConnectorCategoriesCatalog


@dataclass(frozen=True, kw_only=True)
class RepositoryConnectorCategory(ConnectorCategory):
    category_id: typing.ClassVar[ConnectorCategoryID] = "repository"

    def __init__(self):
        super().__init__(
            name="Repository",
            description="Publish your project to make it available to the public.",
        )


ConnectorCategoriesCatalog.register_item(
    RepositoryConnectorCategory.category_id, RepositoryConnectorCategory()
)
